### DISCUSSION TITLE
Quick tips to make Example log some messages in console

### DISCUSSION THREAD

**Discussion by crm-ops** [tdlight-team/tdlight-java#189] [Show and tell]
On top of all readme instructions add the following : 

Example.log4j.xml > replace DEBUG by TRACE

In Example.Main() > add Thread.sleep(100000);

If the Auth mechanism is working ok you'll see all message history in the console

Thanks to @cavallium for this awesome implementation of tdlib ! really powerful :)

**Comment by cavallium** [comment]
Why should I add thread.sleep(10000) in Main?

**Reply by crm-ops** [reply]
Without the Thread.sleep() the example class "autoclose" the client prematurely before to receive the async answer of getChatList update.
Adding this line was the only way for me to see something getting pulled from TG.

**Reply by cavallium** [reply]
The example sends a message and gets the result, then it closes itself. The example is correct. You are trying to personalize the example in a way that requires that modification, it's unnecessary to add it to the original Example.java.

**Reply by crm-ops** [reply]
I might be mistaken - during the execution of the Example, this code (from the AuthorizationStateReadyLoadChats class) 

is triggered somehow 

public void onUpdate(UpdateAuthorizationState update) {
		if (update.authorizationState.getConstructor() == AuthorizationStateReady.CONSTRUCTOR) {
			client.send(new LoadChats(chatList, 10), ok -> {
				if (ok.getConstructor() == Error.CONSTRUCTOR) {
					Error error = (Error) ok;
					if (error.code != 404) {
						throw new TelegramError((Error) ok);
					}
					logger.debug("All {} chats have already been loaded", chatList);
				} else {
					logger.debug("All chats have been loaded");
				}
			}, error -> logger.warn("Failed to execute TdApi.LoadChats()"));
		}
	}


Without the Thread.sleep() => it fails because the client is being closed after the message is sent.

The comments I made are not meant to be pushed to the Master branch of your code - not even a guideline  - simple ideas  that might be helpful for others to run the code Example with no exceptions and getting some of messages update in real time while posting to Telegram channel - for a couple of seconds
